### PR TITLE
Upgrade to jdk 17

### PR DESCRIPTION
**Pull request by calixtus**
Can you try this please on your machine?

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -472,37 +472,20 @@ task copyTestResources(type: Copy) {
...
 
 tasks.withType(Test) {
-    reports.html.destination = file("${reporting.baseDir}/${name}")
+    reports.html.outputLocation.set(file("${reporting.baseDir}/${name}"))
 }
 
-task jacocoMergePrep() {
-    doFirst {
-        // Ignore failures of tests
-        tasks.withType(Test) {
```
Comment (tobiasdiez) [inline build.gradle:481]: Without this the codecoverage workflow is now failing due to failing tests.

Comment (calixtus) [reply build.gradle:481]: I removed it because jacocoMerge is deprecated. We probably have to think of another way to restore the codecoverage workflow.

Comment (calixtus) [reply build.gradle:481]: Codecoverage should work again now...

**Comment by calixtus** [comment]
This PR should not be merged, before we are sure that UNIBAS-SE does not need us to stay at JDK 16 anymore.

**Comment by Siedlerchr** [comment]
We should switch from adopt to temurin, since that is the new name (I hate those name changes) because it might be that the adopt name at one point in the future does no longer work.

https://blog.adoptopenjdk.net/2021/08/goodbye-adoptopenjdk-hello-adoptium/
Otherwise lgtm, the changes work in Eclipse as well

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=b586e04868a22fd817c8971330fec37e298f3242eb85c374181b12d637f80302
```
Comment (koppor) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can't the checksum of the binary be kept?

**Review by koppor — APPROVED** [review]
In general, LGTM - one micro nitpick

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (calixtus) [reply gradle/wrapper/gradle-wrapper.properties:3]: I did not remember removing it.  @Siedlerchr is to blame for https://github.com/JabRef/jabref/pull/8433/commits/5d405cb71991c4737d1399e51581297ef6e2f00e
:wink:

Comment (Siedlerchr) [reply gradle/wrapper/gradle-wrapper.properties:3]: I just updated gradlew to the latest version, there was somehow a merge conflict or whatever

Comment (calixtus) [reply gradle/wrapper/gradle-wrapper.properties:3]: Ok, bit that means the checksum has changed?

Comment (Siedlerchr) [reply gradle/wrapper/gradle-wrapper.properties]: Refs the discussion here: https://github.com/gradle/gradle/issues/12412#issuecomment-653000921
Apparently, when you don't include the parameter it gets deleted.

**Pull request merged**